packages/strapi-plugin-users-permissions/controllers/Auth.js

```diff
@@ -261,7 +261,8 @@ module.exports = {
     }
 
     // Ability to pass OAuth callback dynamically
-    grantConfig[provider].callback = _.get(ctx, 'query.callback') || grantConfig[provider].callback;
+    grantConfig[provider].callback = _.get(ctx, 'query.callback') || _.get(ctx, 'query.state') || grantConfig[provider].callback;
+    grantConfig[provider].state = _.get(ctx, 'query.callback') || grantConfig[provider].callback;  
```

Comment:
Hi, The state might be auto generated by other providers. it seems a bit hacky to set it to the callback value by default. It feels like it will break other providers that use this param.

Reply:
Most likely you might be right. 

I've tested Facebook and Google so far.

Reply:
We might want to explore something like this in grant: https://github.com/simov/grant#dynamic-http